packages/@sanity/studio-hints/src/components/HintsPackage.js

```diff
@@ -0,0 +1,147 @@
+/* eslint-disable no-console, class-methods-use-this */
+import React from 'react'
+import {isEmpty} from 'lodash'
+import Spinner from 'part:@sanity/components/loading/spinner'
+import studioHintsConfig from 'part:@sanity/default-layout/studio-hints-config'
+import WarningIcon from 'part:@sanity/base/warning-icon'
+import {locationSetting, getHints, updateLocation} from '../datastore'
+import {resolveUrl} from './utils'
+import LinksList from './LinksList'
+// import HintPage from './HintPage'
+import styles from './HintsPackage.css'
+
+const removeHintsArticleSlug = 'remove-this-sidebar'
+
+export default class HintsPackage extends React.PureComponent {
+  state = {
+    error: null,
+    hintsPackage: null,
+    sidebarRemovalInstructions: null,
+    activePage: null,
+    isLoading: true
+  }
+
+  subscription = null
+
+  fetchHintsPackage(repoId) {
+    getHints(repoId, removeHintsArticleSlug)
+      .then(result => {
+        const {hintsPackage, sidebarRemovalInstructions} = result
+        this.setState({hintsPackage, sidebarRemovalInstructions, isLoading: false})
+      })
+      .catch(error => this.setState({error, isLoading: false}))
+  }
+
+  componentDidMount() {
+    const repoId = studioHintsConfig.templateRepoId
+    this.fetchHintsPackage(repoId)
+
+    this.subscription = locationSetting.listen().subscribe(currentLocation => {
+      this.setState({activePage: currentLocation ? JSON.parse(currentLocation).id : null})
+    })
+  }
+
+  componentWillUnmount() {
+    if (this.subscription) {
+      this.subscription.unsubscribe()
+    }
+  }
+
+  /**
```

Comment:
Any chance we can remove the commented out code here, or keep it functional in a separate branch? Commented out code quickly get's outdated. I'd rather leave a detailed explanation of the problem and prefer pseudo-code over actual code in these cases.

Reply:
I very much agree. I want it out too, but both @jorngeorg and @vicbergquist said _we want it in there because it's coming back in really soon_